api/internal/plugins/loader/loader.go

```diff
@@ -225,9 +225,26 @@ func (l *Loader) makeBuiltinPlugin(r resid.Gvk) (resmap.Configurable, error) {
 	return nil, errors.Errorf("unable to load builtin %s", r)
 }
 
-func (l *Loader) loadPlugin(res *resource.Resource) (resmap.Configurable, error) {
+func (l *Loader) loadPlugin(ldr ifc.Loader, res *resource.Resource) (resmap.Configurable, error) {
 	spec := fnplugin.GetFunctionSpec(res)
 	if spec != nil {
+		// validation check that function mounts are under the current kustomization directory
+		for _, mount := range spec.Container.StorageMounts {
+			var relPath string
+			var err error
+			if filepath.IsAbs(mount.Src) {
```

Comment:
WDYT about disallowing absolute paths entirely, on the grounds of them breaking encapsulation, as you pointed out?

Reply:
done!